### PR TITLE
ci: enable IPv6 test

### PR DESCRIPTION
**Pull request by zirain**

### CODE REVIEW

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -122,8 +122,8 @@ jobs:
         - version: v1.30.4
           ipFamily: ipv4
         # Enable these after https://github.com/envoyproxy/gateway/issues/4572 fixed
-        # - version: v1.31.0
-        #   ipFamily: ipv6   # only run ipv6 test on latest version to save time
+        - version: v1.31.0
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:125]: can we do the for v1.30 so we save some time ?

File: tools/make/kube.mk [hunk 2]
```diff
@@ -40,6 +40,17 @@ endif
 YEAR := $(shell date +%Y)
 CONTROLLERGEN_OBJECT_FLAGS :=  object:headerFile="$(ROOT_DIR)/tools/boilerplate/boilerplate.generatego.txt",year=$(YEAR)
 
+.PHONY: prepare-ip-family
+prepare-ip-family: ## Prepare the environment for running e2e tests with ENVOY_PROXY_IP_FAMILY
+	@$(LOG_TARGET)
+ifeq ($(IP_FAMILY),ipv4)
```
Comment (arkodg) [inline tools/make/kube.mk:46]: can we just reuse the EG enums in make file and scripts to reduce cognitive complexity ?

File: test/benchmark/config/gatewayclass.yaml [hunk 3]
```diff
@@ -16,6 +16,7 @@ metadata:
   name: proxy-config
   namespace: envoy-gateway-system
 spec:
+  ipFamily: $ENVOY_PROXY_IP_FAMILY
```
Comment (arkodg) [inline test/benchmark/config/gatewayclass.yaml:19]: neat idea :)

File: .github/workflows/build_and_test.yaml (hunk 1 above)
Comment (zirain) [reply .github/workflows/build_and_test.yaml:125]: sure

File: tools/make/kube.mk (hunk 2 above)
Comment (zirain) [reply tools/make/kube.mk:46]: `kind` use `ipv4` `ipv6` and `dual`, which is different with EG.

Comment (arkodg) [reply tools/make/kube.mk:46]: ah cool

File: test/benchmark/config/gatewayclass.yaml (hunk 3 above)
Comment (zirain) [reply test/benchmark/config/gatewayclass.yaml:19]: the bad news is that we need move all EnvoyProxy resources to this file.

**Comment by zirain** [comment]
https://github.com/envoyproxy/gateway/pull/4853 is better

**Pull request closed without merging**